narwhals/functions.py

```diff
@@ -1040,6 +1040,42 @@ def func(plx: Any) -> Any:
     return Expr(func, ExprMetadata.selector())
 
 
+def exclude(*names: str | Iterable[str]) -> Expr:
+    """Creates an expression that excludes columns by their name(s).
+
+    Arguments:
+        names: Name(s) of the columns to exclude.
+
+    Returns:
+        A new expression.
+
+    Examples:
+        >>> import polars as pl
+        >>> import narwhals as nw
+        >>>
+        >>> df_native = pl.DataFrame({"a": [1, 2], "b": [3, 4], "c": ["x", "z"]})
+        >>> nw.from_native(df_native).select(nw.exclude("c", "a"))
+        ┌──────────────────┐
+        |Narwhals DataFrame|
+        |------------------|
+        |  shape: (2, 1)   |
+        |  ┌─────┐         |
+        |  │ b   │         |
+        |  │ --- │         |
+        |  │ i64 │         |
+        |  ╞═════╡         |
+        |  │ 3   │         |
+        |  │ 4   │         |
+        |  └─────┘         |
+        └──────────────────┘
+    """
+
```

Comment:
https://github.com/narwhals-dev/narwhals/pull/2122/files/d733bb24b79c1f79f3033b8433e82f76cf056c1d#r1976440039

Defining it here would save the cost of re-evaluating on each call
```suggestion
    exclude_names = frozenset(flatten(column_names))

```

Then passing that to `plx.exclude(...)`

Reply:
Resolved via (https://github.com/narwhals-dev/narwhals/pull/2122#discussion_r1976480393)